### ISSUE TITLE
Jumping into middle of instruction causes wrong disassembly

### ISSUE THREAD

**Issue by pinobatch** [SourMesen/Mesen#513]
As [never-obsolete discovered](https://forums.nesdev.com/viewtopic.php?p=226977#p226977), Mesen's debugger doesn't like jumping into instructions.
```
;;
; @param cellpos position to read
; @return A, v0: value at that position, or $FF if out of range
read_cell:
  ldx cellpos
  cpx #NUM_CELLS
  bcs _out_of_range
    lda array,x
    .byte $2C      ; first byte of BIT $FFA9
  _out_of_range:
    lda #$FF
  sta v0
  rts
```
The first time it ran, `cellpos` was in range, so it went through the array access path (`lda, bit, sta, rts`).

Later on, `cellpos` was out of range, so the branch was taken, and it went through the constant access path (`lda, sta, rts`). While stepping through, the debugger branched to the line `BIT $FFA9`, but executed `LDA #$FF`.

When PC happens to be at `LDA #$FF`, the debugger needs to show whatever instruction the PC points to, even if a previous instruction's disassembly has to be amended to a `.BYTE`.

**Comment by SourMesen** [comment]
Thanks - this should be fixed by 70ad89a05a182a9ca290c5327d00db106d2eb9db.

e.g with this change, if you have:
```
$8000    LDA #$EA
$8002    STA $2000
```
but make a jump to $8001, the disassembly should now read:
```
$8000    LDA #$EA
$8001    NOP
$8002    STA $2000
```

Whereas it would previously highlight the $8000 line and pretend it was executing the LDA operation, even though the emulator was actually (correctly) running the NOP.

**Comment by pinobatch** [comment]
[Notified downstream reporter of fix](https://forums.nesdev.com/viewtopic.php?p=230139#p230139)
